lib/tasks/import_projects_task.rb

```diff
@@ -14,22 +14,62 @@ def sync
         pp airtable_project
         puts ("-" * 20)
         proj = Project.find_or_initialize_by name: airtable_project[:project_name]
-        proj.leads = User.where(:slack_userid => airtable_project[:project_lead_slack_id]) unless airtable_project[:project_lead_slack_id].blank?
-        proj.volunteers += User.where(:slack_userid => airtable_project[:team_member_ids]) unless airtable_project[:team_member_ids].blank?
-        proj.skills = Skill.match_skills(airtable_project[:tech_stack]).tech_skills unless airtable_project[:tech_stack].blank?
+
+
+        airtable_project[:project_lead_slack_id].each do |slack_id|
+          lead = User.find_by(:slack_userid => slack_id)
+          if lead != nil
+            unless proj.lead_ids.include?(lead.id)
+              proj.lead_ids << lead.id
+            end
+          end
+        end unless airtable_project[:project_lead_slack_id].blank?
+
+        airtable_project[:team_member_ids].each do |member_id|
+          volunteer = User.find_by(:slack_userid => member_id)
+          if volunteer != nil
+            unless proj.volunteers.include?(volunteer)
+              proj.volunteers << volunteer
+            end
+          end
+        end unless airtable_project[:team_member_ids].blank?
+
+        airtable_project[:progcode_coordinator_ids].each do |coord_id|
+           coordinator = User.find_by(:slack_userid => coord_id)
+           if coordinator != nil
+             unless proj.progcode_coordinator_ids.include?(coordinator.id)
+            proj.progcode_coordinator_ids << coordinator.id
+           end
+          end
+        end unless airtable_project[:progcode_coordinator_ids].blank?
+
+        airtable_project[:needs_categories].each do |category|
+          skill = Skill.where('lower(name) = ?', category.downcase).first_or_create(:name=>category)
+          unless proj.needs_categories.include?(skill)
```

Comment:
hmm, these are very weird calls.  Why doesnt the `unless` block have  a `do` statement?

these should look like:

```
airtable_project[:progcode_coordinator_ids].each do |coord_id|
    coordinator = User.find_by(:slack_userid => coord_id)
    proj.progcode_coordinator_ids << coordinator.try(:id)

    # this will remove nils and duplicates
    proj.progcode_coordinator_ids = proj.progcode_coordinator_ids.uniq.compact 
end unless airtable_project[:progcode_coordinator_ids].blank?
```

Reply:
oh yeah, because a nonexistent ID will return a blank relation, not nil

Reply:
new commit completely changed this workflow because nil values are occasionally tracked and lead to error messages. for columns where nil value was acceptable i adopted this structure.